### PR TITLE
Remove android directory from platform_channel_swift example project

### LINKED ISSUE #49619: Remove Unused Imports and  Migrate examples to Android embedding v2
## Use case
When I am checking the platform_channel_swift example project, I see that MainActivity is not Migrated to Android embedding v2. Also, I want to remove the unused imports in the MainActivity class. 

## Proposal

I have fix for the above use case. Can I raise the Pull Request?

### PR DESCRIPTION
**Pull request by jmagman**
## Description
Android directory added in https://github.com/flutter/flutter/pull/35778 has atrophied.  The purpose of this example project is to emphasize the Swift code, so remove the `android` directory so it can't be run. 

Change SHARD=build_tests integration test to skip Android or iOS tests if not supported.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/49619.

## Tests

`platform_channel_sample_test_swift` devicelab task should still pass.

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Review by jamesderlin — APPROVED** [review]

File: dev/bots/test.dart [hunk 1]
```diff
@@ -282,13 +282,6 @@ Future<void> _runToolTests() async {
   await selectSubshard(subshards);
 }
 
-// Example apps that should not be built by _runBuildTests`
-const List<String> _excludedExampleApplications = <String>[
```
Comment (jmagman) [inline dev/bots/test.dart]: \cc @jonahwilliams

Comment (jmagman) [reply dev/bots/test.dart]:
```
Example project splash has no android directory, skipping aot and apk
Example project splash has no ios directory, skipping ipa
```

Comment (jonahwilliams) [reply dev/bots/test.dart]: SGTM

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
Landed on red, fuchsia infra failure.

**Pull request merged**